### PR TITLE
Node project new branch

### PR DESCRIPTION
**Pull request by shine377**

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module.exports = {
+  development: {
+    username: process.env.USER_NAME,
+    password: process.env.PASSWORD,
+    database: process.env.DATABASE,
+    host: process.env.HOST,
+    dialect: process.env.DIALECT,
+  },
+  test: {
+    username: "root",
```
Comment (kirubasunder) [inline config/config.js]: test environment database details follows the same pattern as development. isn't it? Please update this same as development.

File: config/config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    password: null,
+    database: "database_test",
+    host: "127.0.0.1",
+    dialect: "mysql",
+  },
+  production: {
+    username: "root",
```
Comment (kirubasunder) [inline config/config.js]: production environment database details follows the same pattern as development. isn't it? Please update this same as development.

File: controllers/assessment-controllers.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+const assessment = require("../models/assessmentDetail");
+const studentData = require("../models/studentModal");
+const teacherData = require("../models/teacherModal");
+
+
+//Getting data from students table
+const tableUser = async (req, res) => {
+  let data = await studentData.findAll({});
+  try {
+    console.log(data);
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Application should not have console logs. Log are maintained as logger with levels. So delete the console logs.

File: controllers/assessment-controllers.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+const assessment = require("../models/assessmentDetail");
+const studentData = require("../models/studentModal");
+const teacherData = require("../models/teacherModal");
+
+
+//Getting data from students table
+const tableUser = async (req, res) => {
+  let data = await studentData.findAll({});
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Use self explaining variable name.

File: controllers/assessment-controllers.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  try {
+    console.log(data);
+
+    res.send(data);
+    console.log("Got the response");
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Application should not have console logs. Log are maintained as logger with levels. So delete the console logs.

File: controllers/assessment-controllers.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  } catch (err) {
+    console.log("user not found", err);
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Catch should return error type instead it is just console it. Please update it

File: controllers/assessment-controllers.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  }
+};
+
+
+//Posting data from the form to the table
+ const updateTableUser = async (req, res) => {
+  let data = {
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Use self explaining variable name.

File: controllers/assessment-controllers.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    approvedDueDate: req.body.approvedDueDate,
+    extendedBy: req.body.extendedBy,
+    Status: "pending",
+    teacherId: 1,
+    studentName: "Shiny",
+  };
+  const user = await studentData.create(data);
+  try {
+    res.send(user);
+    console.log("user created successfully");
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Application should not have console logs. Log are maintained as logger with levels. So delete the console logs.

File: controllers/assessment-controllers.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  } catch (err) {
+    console.log("User not created", err);
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Catch should return error type instead it is just console it. Please update it

File: controllers/assessment-controllers.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+//Fetching all the assessments in the dropdown
+const getAllUser = async (req, res) => {
+  let data = await assessment.findAll({});
+  try {
+    console.log(data);
+
+    res.send(data);
+    console.log("Got the response");
+  } catch (err) {
+    console.log("user not found", err);
```
Comment (kirubasunder) [inline controllers/assessment-controllers.js]: Catch should return error type instead it is just console it. Please update it

File: models/index.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  });
+
+Object.keys(db).forEach(modelName => {
+  if (db[modelName].associate) {
+    db[modelName].associate(db);
+  }
+});
+
+db.sequelize = sequelize;
+db.Sequelize = Sequelize;
```
Comment (kirubasunder) [inline models/index.js:41]: why it is declared twice?

File: utils/index.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  .sync({ alter: true })
+  .then((result) => {
+    console.log("created");
+  })
+  .catch((err) => {
+    console.log("error occured", err);
+  });
+
+app.listen(port, () => {
+  console.log("Running on port 3000");
```
Comment (kirubasunder) [inline utils/index.js]: mislead by port number. App runs at 8000

File: utils/index.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+const router = require("../routers/user-router");
+const sequelize = require("../database/creatingConnection");
+
+app.use(cors());
+
+app.use(express.json());
+
+app.use(express.urlencoded({ extended: true }));
+
+app.use("/api/v10", router);
```
Comment (kirubasunder) [inline utils/index.js]: should be `api/v1`

File: utils/index.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  .then((result) => {
+    console.log("created");
+  })
+  .catch((err) => {
+    console.log("error occured", err);
+  });
+
+app.listen(port, () => {
+  console.log("Running on port 3000");
+});
```
Comment (kirubasunder) [inline utils/index.js]: this not the util file, the app init file itself. This file should be under the root

File: package-lock.json
Comment (kirubasunder) [inline package-lock.json:1]: If the application uses `yarn` as a package manager. Please delete this file

File: config/config.js
Comment (kirubasunder) [inline config/config.js:1]: rename file to be more specific. `config/config.js` doesn't mean anything

File: controllers/assessment-controllers.js
Comment (kirubasunder) [inline controllers/assessment-controllers.js:1]: No controller string in the file name. As it is already under the controllers folder itself.

File: routers/user-router.js
Comment (kirubasunder) [inline routers/user-router.js:1]: No route string in the file name. As it is already under the routers folder itself.

File: config/config.js (hunk 1 above)
Comment (shine377) [reply config/config.js]: Updated the same details as development.

File: config/config.js (hunk 2 above)
Comment (shine377) [reply config/config.js]: updated the same details as development

File: controllers/assessment-controllers.js (hunk 3 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: Deleted the console log

File: controllers/assessment-controllers.js (hunk 5 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: Deleted the console log

File: controllers/assessment-controllers.js (hunk 6 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: updated the catch block with return error type

File: controllers/assessment-controllers.js (hunk 8 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: console log was deleted

File: controllers/assessment-controllers.js (hunk 9 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: updated the catch block with return error type

File: controllers/assessment-controllers.js (hunk 10 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: updated the catch block with return error type

File: utils/index.js (hunk 12 above)
Comment (shine377) [reply utils/index.js]: Changes the port number as 8000

File: utils/index.js (hunk 13 above)
Comment (shine377) [reply utils/index.js]: Changes to api/v1

File: package-lock.json
Comment (shine377) [reply package-lock.json:1]: yarn lock file was deleted

File: utils/index.js (hunk 14 above)
Comment (shine377) [reply utils/index.js]: Changed the index.js file under root

File: controllers/assessment-controllers.js (hunk 4 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: Variable name is changed

File: controllers/assessment-controllers.js (hunk 7 above)
Comment (shine377) [reply controllers/assessment-controllers.js]: variable name is changed

File: routers/user-router.js
Comment (shine377) [reply routers/user-router.js:1]: The route string in the file name is removed

File: controllers/assessment-controllers.js
Comment (shine377) [reply controllers/assessment-controllers.js:1]: The controller string in the file name is removed

File: config/config.js
Comment (shine377) [reply config/config.js:1]: changed the file name

File: models/index.js (hunk 11 above)
Comment (shine377) [reply models/index.js:41]: The index.js file is pre-defined by sequelize. without these declaration its is not possible to run the project

**Pull request merged**